### PR TITLE
Dev dispatch

### PR DESCRIPTION
**Pull request by yuiokubo**
add 2 priority rules ( VC and FIFO )

### CODE REVIEW

File: pDESy/model/base_priority_rule.py [hunk 1]
```diff
@@ -18,6 +21,8 @@ class TaskPriorityRuleMode(IntEnum):
     EST = 1  # Earliest Start Time
     SPT = 2  # Shortest Processing Time
     LPT = 3  # Longest Processing Time
+    FIF0 = 4 # First in First out
```
Comment (taiga4112) [inline pDESy/model/base_priority_rule.py:24]: Change `FIF0` to `FIFO`

File: pDESy/model/base_priority_rule.py [hunk 2]
```diff
@@ -82,5 +93,10 @@ def sort_task_list(task_list, priority_rule_mode=TaskPriorityRuleMode.TSLACK):
             task_list = sorted(
                 task_list, key=lambda task: task.default_work_amount, reverse=True
             )
+        elif priority_rule_mode == TaskPriorityRuleMode.FIFO:
+            # Task: FIFO (First In First Out rule)
+            task_list = sorted(
+                task_list, key=lambda task: task.state_record_list.count(1),reverse=True
```
Comment (taiga4112) [inline pDESy/model/base_priority_rule.py:99]: Please use `BaseTaskState.READY`, not `1` or other numerical.

File: tests/model/test_base_priority_rule.py [hunk 3]
```diff
@@ -64,6 +65,22 @@ def test_sort_task_list_LPT():
     assert task_list[1].name == "t1"
     assert task_list[2].name == "t0"
 
+def test_sort_task_list_FIFO():
+    t0 = BaseTask("t0")
+    t0.state_record_list=[1,1,1]
```
Comment (taiga4112) [inline tests/model/test_base_priority_rule.py:70]: Please use `BaseTaskState.xxx`, not `0` ,`1` and other numbers.

**Review by taiga4112 — CHANGES_REQUESTED** [review]
Please check my review comment.

File: pDESy/model/base_priority_rule.py (hunk 2 above)
Comment (yuiokubo) [reply pDESy/model/base_priority_rule.py:99]: > 
```suggestion
                task_list, key=lambda task: task.state_record_list.count(BaseTaskState.READY),reverse=True
```

error code 
`AILED tests/model/test_base_priority_rule.py::test_sort_task_list_FIFO - NameError: name 'BaseTaskState' is not defined`

**Review by taiga4112 — APPROVED** [review]

**Pull request merged**